src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriter.cs

```diff
@@ -460,7 +426,7 @@ public virtual void WriteNode(XmlReader reader, bool defattr)
                         WriteFullEndElement();
                         break;
                 }
-            } while (reader.Read() && (d < reader.Depth || (d == reader.Depth && reader.NodeType == XmlNodeType.EndElement)));
+            } while (reader.Read() && (d < reader.Depth || d == reader.Depth && reader.NodeType == XmlNodeType.EndElement));
```

Comment:
Please revert this.  It's not clear to most readers that && has higher precedence than ||.